tests/estimators/certification/test_smooth_vit.py

```diff
@@ -0,0 +1,745 @@
+
+    elif ablation == "block":
+        ablator = BlockAblator(
+            ablation_size=4,
+            channels_first=True,
+            to_reshape=True,
+            original_shape=(3, 32, 32),
+            output_shape=(3, 224, 224),
+            mode="ViT",
+        )
+        ablated = ablator.forward(cifar_data, column_pos=10, row_pos=28)
+        madry_preds = madry_vit(ablated)
+        art_preds = art_model.model(ablated)
+        assert torch.allclose(madry_preds, art_preds, rtol=1e-04, atol=1e-04)
+
+    sys.path.remove("smoothed-vit/src/utils/")
+
+
+@pytest.mark.only_with_platform("pytorch")
+@pytest.mark.parametrize("ablation", ["block", "column"])
+def test_certification_equivalence(art_warning, fix_get_mnist_data, fix_get_cifar10_data, ablation):
+    """
+    With the forward pass equivalence asserted, we now confirm that the certification functions in the same
+    way by doing a full end to end prediction and certification test over the data.
+    """
+    import torch
+    import os
+    import sys
+    import types
+
+    from torch.utils.data import Dataset
+
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    experiment_file_directory = "smooth_vit_tests"
+
+    class ArgClass:
+        def __init__(self):
+            self.certify_patch_size = 4
+            self.certify_ablation_size = 4
+            self.certify_stride = 1
+            self.dataset = "cifar10"
+            self.certify_out_dir = "./"
+            self.exp_name = experiment_file_directory
+            if ablation == "column":
+                self.certify_mode = "col"
+            if ablation == "block":
+                self.certify_mode = "block"
+            self.batch_id = None
+
+    class DataSet(Dataset):
+        def __init__(self, x, y):
+            self.x = x
+            self.y = y
+
+        def __len__(self):
+            return len(self.y)
+
+        def __getitem__(self, idx):
+            return self.x[idx], self.y[idx]
+
+    from art.estimators.certification.derandomized_smoothing import PyTorchDeRandomizedSmoothing
+    import shutil
+    from torch.utils.data import DataLoader
+
+    if os.path.exists(experiment_file_directory):
+        shutil.rmtree(experiment_file_directory)
+
+    os.system("git clone https://github.com/MadryLab/smoothed-vit")
+    sys.path.append("smoothed-vit/src/utils/")
+    from smoothing import certify
+
+    art_model = PyTorchDeRandomizedSmoothing(
+        model="vit_small_patch16_224",
+        loss=torch.nn.CrossEntropyLoss(),
+        optimizer=torch.optim.SGD,
+        optimizer_params={"lr": 0.01},
+        input_shape=(3, 32, 32),
+        nb_classes=10,
+        ablation_type=ablation,
+        ablation_size=4,
+        load_pretrained=True,
+        replace_last_layer=True,
+        verbose=False,
+    )
+
+    # TODO: Look into incorporating this model into the CI runs rather than just local testing.
+    if os.path.isfile("vit_small_patch16_224_block.pt"):
```

Comment:
Where is the location of this file?

Reply:
to avoid having a large ~87MB pertained model weights as part of the PR, we have refactored this to only load a pre-trained set of weights corresponding to the output layer.